posydon/binary_evol/CE/step_CEE.py

```diff
@@ -540,364 +952,468 @@ def CEE_simple_alpha_prescription(
+        """Update the binary and component stars upon merging within a CEE.
 
-                if (Mejected_donor > m1_i  - mc1_i) or (Mejected_comp > m2_i  - mc2_i):
-                    Mejected_donor = (m1_i  - mc1_i) -0.01 # at least this value of envelope is left.
-                    Mejected_comp = (m2_i  - mc2_i) -0.01
-                    Pwarn("M_ejected of at least one star in double CEE is found to be more than the initial envelope. "
-                                  "Reducing both to their initial_envelope - 0.01 Msun", "ApproximationWarning")
+        The binary's state and event are updated along with the donor and 
+        companion star masses and radii corresponding to a merger event.
 
-            donor.mass = m1_i - Mejected_donor
-            donor.log_R = np.nan
-            comp_star.mass = m2_i - Mejected_comp
-            comp_star.log_R = np.nan
-            if donor_type == 'CO_core':
-                donor.he_core_mass = m1_i - Mejected_donor
-                donor.he_core_radius = np.nan
-                comp_star.he_core_mass = m2_i - Mejected_comp
-                comp_star.he_core_radius = np.nan
+        Parameters
+        ----------
+        binary: BinaryStar object
+            The binary system
+        donor : SingleStar object
+            The donor star
+        comp_star : SingleStar object
+            The companion star
+        m1_i : float
+            mass of the donor upon entering a CE (in Msun)
+        m2_i : float
+            mass of the companion upon entering a CE (in Msun)
+        donor_type : string
+            descriptor for the stellar type of the donor
+        Mejected_donor : float
+            how much mass is ejected from the donor upon merger (in Msun)
+        Mejected_comp : float
+            how much mass is ejected from the companion upon merger (in Msun)
+        verbose : bool
+            In case we want information about the CEE.
+        """
+        # system merges
+        binary.state = 'merged'
+        if binary.event in ["oCE1", "oDoubleCE1"]:
+            binary.event = "oMerging1"
+        if binary.event in ["oCE2", "oDoubleCE2"]:
+            binary.event = "oMerging2"
 
-            if verbose:
-                print("The mass loss during merging_CEE is: ", mass_loss_during_CEE_merged)
-                print("so m1_i , m1_f = ", m1_i, donor.mass)
-                print("so m2_i , m2_f = ", m2_i, comp_star.mass)
+        if verbose:
+            print("system merges due to one of the two star's core filling"
+                    "its RL")
+
+        donor.mass = m1_i - Mejected_donor
+        donor.log_R = np.nan
+        comp_star.mass = m2_i - Mejected_comp
+        comp_star.log_R = np.nan
+        if donor_type == 'CO_core':
+            donor.he_core_mass = m1_i - Mejected_donor
+            donor.he_core_radius = np.nan
+            comp_star.he_core_mass = m2_i - Mejected_comp
+            comp_star.he_core_radius = np.nan
```

Comment:
This required to add the comp_type to the input parameters.
```suggestion
        if comp_type == 'CO_core':
            comp_star.he_core_mass = m2_i - Mejected_comp
            comp_star.he_core_radius = np.nan
```